### DISCUSSION TITLE
No coverage for deploy script MoodNft

### DISCUSSION THREAD

**Discussion by jaimebarrancos** [Cyfrin/foundry-full-course-cu#283] [Q&A] [answered]
I have a few integration tests that deploy and interact with the MoodNft contract at:
https://github.com/jaimebarrancos/foundry-nft/blob/master/test/integrations/MoodNftIntegrationTest.t.sol

Running ```forge coverage``` I get:
 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/90402548/b246c0d3-4a02-4806-931b-6eb882d5e69a)

But I don't understand why my DeployMoodNft script has 0 % coverage. I thought my tests covered at least a portion of it? Also how do I increase the coverage and make a great test for the deployment script, thanks!

**Comment by shikhar229169** [comment]
Actually you didn't wrote any test for DeployMoodNft.
So what you can do is test the function which converts svg image to URI and also test the run function of the DeployMoodNft, whether the token URI are set up correctly or not.

**Comment by f3d0ss** [comment, marked as answer]
You don't get coverage on the `DeployMoodNft` because you are executing his `run()` function only in the ` function setUp()` of the test file and anything that is executed in that function do not count in the coverage.
If you would run the following code in one of the tests it should probably increase the `DeployMoodNft` coverage.
``` solidity
deployer = new DeployMoodNft();
moodNft = deployer.run();
```


Anyway you should test the `DeployMoodNft` in a separate test file dedicated to that contract
